src/Interface/Modules/Render/GLWidget.cc

```diff
@@ -70,12 +70,20 @@ GLWidget::~GLWidget()
 void GLWidget::initializeGL()
 {
 	spire::glPlatformInit();
-}	
+}
 
 void GLWidget::paintGL()
 {
-  mCurrentTime += updateTime;
-  mGraphics->doFrame(mCurrentTime, updateTime);
+  //set to 200ms to force promise fullfilment every frame if a good frame as been requested
```

Comment:
How does this affect framerate? Or does it?

Reply:
This will only happen on execution and execution will halt until we get a "good frame" so whether or not this effects frame rate doesn't really matter. What it does do is accelerate execution.

Reply:
An interesting test case then is animation using GetMatrixSlice, which triggers many executions in a row.